src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -1091,6 +1091,9 @@ export function getFormattedType(program: Program, type: Type): FormattedType {
     const format = getFormat(program, type);
     if (type.kind === "ModelProperty") {
         targetType = type.type;
+        if (targetType.kind === "UnionVariant") {
```

Comment:
we may need to add this new switch case in `getInputType`, when kind == `UnionVariant`, we use union variant type.
In that way we will support more case which use a UnionVariant as type, not just model property.